src/Godot.SourceGenerators/EntryPointGenerator.cs

```diff
@@ -25,12 +27,56 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             .Collect()
             .Select((topLevelAttrs, ct) => !topLevelAttrs.IsEmpty);
 
-        IncrementalValueProvider<ImmutableArray<GodotRegistrationSpec?>> registrationSpecs = context.SyntaxProvider
+        IncrementalValueProvider<ImmutableArray<GodotRegistrationSpec>> registrationSpecs = context.SyntaxProvider
             .ForAttributeWithMetadataName(KnownTypeNames.GodotClassAttribute,
                 predicate: IsSyntaxTargetForGeneration,
                 transform: GetSemanticTargetForGeneration)
             .Where(spec => spec is not null)
-            .Collect();
+            .Collect()
+            //sort for parent classes to to be registered before child classes
+            .Select((spec, ct) =>
+            {
+                Dictionary<string, GodotRegistrationSpec> parentTree = [];
+                Dictionary<string, int> dependencyTree = [];
+
+                foreach (var nullable in spec)
+                {
+                    var type = nullable!.Value;
+
+                    parentTree.Add(type.FullyQualifiedSymbolName, type);
+
+                    int dependencies = 1;
+                    if (dependencyTree.TryGetValue(type.FullyQualifiedSymbolName, out var children))
+                    {
+                        dependencies = children + 1;
+                    }
+                    else
+                    {
+                        dependencyTree.Add(type.FullyQualifiedSymbolName, 0);
+                    }
+
+                    string parent = type.FullyQualifiedParentName;
+                    while (parentTree.TryGetValue(parent, out var parentType))
+                    {
+                        dependencyTree[parent] += dependencies;
+                        parent = parentType.FullyQualifiedParentName;
+                    }
+
+                    if (dependencyTree.ContainsKey(parent))
+                    {
+                        dependencyTree[parent] += dependencies;
+                    }
+                    else
+                    {
+                        dependencyTree.Add(parent, dependencies);
+                    }
```

Comment:
Since we always end up assigning to the dictionary, we could reduce it to this:

```suggestion
                    derivedTypeCountByBaseTypeName.TryGetValue(baseTypeName, out int oldDerivedTypeCount);
                    derivedTypeCountByBaseTypeName[baseTypeName] = oldDerivedTypeCount + derivedTypeCount;
```

Although I'm not sure it's better.